src/gui/src/gui.i

```diff
@@ -54,123 +99,132 @@ bool enabled()
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 13, "Command selection_add_net is not usable in non-GUI mode");
+  if (!check_gui("selection_add_net")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addSelectedNet(name);
 }
 
 void
 selection_add_nets(const char* name)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 5, "Command selection_add_nets is not usable in non-GUI mode");
+  if (!check_gui("selection_add_nets")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addSelectedNets(name);
 }
 
 void
 selection_add_inst(const char* name)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 6, "Command selection_add_inst is not usable in non-GUI mode");
+  if (!check_gui("selection_add_inst")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addSelectedInst(name);
 }
 
 void
 selection_add_insts(const char* name)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 7, "Command selection_add_insts is not usable in non-GUI mode");
+  if (!check_gui("selection_add_insts")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addSelectedInsts(name);
 }
 
 void highlight_inst(const char* name, int highlightGroup)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 8, "Command highlight_inst is not usable in non-GUI mode");
+  if (!check_gui("highlight_inst")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addInstToHighlightSet(name, highlightGroup);
 }
 
 void highlight_net(const char* name, int highlightGroup=0)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 9, "Command highlight_net is not usable in non-GUI mode");
+  if (!check_gui("highlight_net")) {
     return;
   }
+  auto gui = gui::Gui::get();
   gui->addNetToHighlightSet(name, highlightGroup);
 }
 
-void add_ruler(int x0, int y0, int x1, int y1)
+void add_ruler(double x0, double y0, double x1, double y1)
 {
-  auto gui = gui::Gui::get();
-  if (!gui) {
-    ord::OpenRoad::openRoad()->getLogger()->info(GUI, 10, "Command add_ruler is not usable in non-GUI mode");
-    return ;
+  if (!check_gui("add_ruler")) {
+    return;
   }
-  gui->addRuler(x0, y0, x1, y1);  
+  odb::Point ll = make_point(x0, y0);
+  odb::Point ur = make_point(x1, y1);
+  auto gui = gui::Gui::get();
+  gui->addRuler(ll.x(), ll.y(), ur.x(), ur.y());  
 }
 
-// converts from microns to DBU
 void zoom_to(double xlo, double ylo, double xhi, double yhi)
 {
-  auto gui = gui::Gui::get();
-  auto logger = ord::OpenRoad::openRoad()->getLogger();
-  if (!gui) {
-    logger->info(GUI, 11, "Command zoom_to is not usable in non-GUI mode");
-    return ;
+  if (!check_gui("zoom_to")) {
+    return;
   }
-  auto db = ord::OpenRoad::openRoad()->getDb();
-  if (!db) {
-    logger->error(GUI, 1, "No database loaded");
```

Comment:
This used to check that the db was loaded.  I think check_gui should probably do that too as not many gui commands make sense if no db is loaded.